scripts/export_costs_from_partial_run.py

```diff
@@ -0,0 +1,173 @@
+        values = []
+        for it in result.plan.itineraries:
+            val = getattr(it, s)
+            # Set value to NaN if it doesn't exist or isn't set
+            if val is None:
+                val = np.nan
+            values.append(val)
+
+        matrix_values[f"mean_{s}"] = np.nanmean(values)
+
+        if matrix_values["number_itineraries"] > 1:
+            matrix_values[f"median_{s}"] = np.nanmedian(values)
+            matrix_values[f"min_{s}"] = np.nanmin(values)
+            matrix_values[f"max_{s}"] = np.nanmax(values)
+            matrix_values[f"num_nans_{s}"] = np.sum(np.isnan(values))
+
+    matrix_values["min_startTime"] = min(i.startTime for i in result.plan.itineraries)
+    matrix_values["max_startTime"] = max(i.startTime for i in result.plan.itineraries)
+    matrix_values["min_endTime"] = min(i.endTime for i in result.plan.itineraries)
+    matrix_values["max_endTime"] = max(i.endTime for i in result.plan.itineraries)
+
+    return matrix_values
+
+
+def _write_matrix_files(
+        data: list[dict[str, float]],
+        matrix_file: pathlib.Path,
+) -> None:
+    matrix = pd.DataFrame(data)
+
+    if type(matrix_file) == str:
+        matrix_file = Path(matrix_file)
+
+    metrics_file = matrix_file.with_name(matrix_file.stem + "-metrics.csv")
+    matrix.to_csv(metrics_file, index=False)
+    print("Written cost metrics to {}\nThis can be used to create a lookup of trips requested".format(metrics_file))
+
+
+def mapcount(filename):
+    """
+    Function to quickly determine response file length
+    https://stackoverflow.com/questions/845058/how-to-get-line-count-of-a-large-file-cheaply-in-python
+    """
+    with open(filename, "r+") as f:
+        buf = mmap.mmap(f.fileno(), 0)
+        lines = 0
+        readline = buf.readline
+        while readline():
+            lines += 1
+        return lines
+
+
+def cost_matrix_from_responses(
+        responses_file: pathlib.Path,
+        matrix_file: pathlib.Path,
+) -> None:
+    """Create cost matrix CSV from responses JSON lines file.
+
+    Parameters
+    ----------
+    responses_file : pathlib.Path
+        Path to JSON lines file containing `CostResults`.
+    matrix_file : pathlib.Path
+        Path to CSV file to output cost metrics to.
+    """
+    matrix_data = []
+    with open(responses_file, "rt") as responses:
+
+        print("Determining response file length")
+        # Get length of response file & report to user
+        response_length = mapcount(responses_file)
+        print("Evaluating {} response lines. This may take some time...".format(response_length))
+
+        for line in tqdm.tqdm(
+                responses, desc="Calculating cost matrix for", dynamic_ncols=True
+        ):
+            results = CostResults.parse_raw(line)
+            # TODO(MB) Recalculate generalised cost if new parameters are provided
+            matrix_data.append(_matrix_costs(results))
+
+    _write_matrix_files(matrix_data, matrix_file)
+
```

Comment:
Import `cost` module instead of copying code

Reply:
Addressed in [2bcb0cb](https://github.com/Transport-for-the-North/otp4gb-py/commit/2bcb0cbeab8c5102221a03f1a7c77665e5435e4e). See https://github.com/Transport-for-the-North/otp4gb-py/pull/4#discussion_r1255600822